### PR TITLE
Appliance management

### PR DESCRIPTION
**Pull request by Misenko**
Implements appliance synchronization process - decides which appliances/image lists will be removed/added/updated.

Since this is kind of a gluing component for all the other pieces, changes are made throughout the whole project. Sorry for the longer pull request.

Proper error handling will be done in the next pull request. Tests will be added after at least some backend development so I can record the real communication.

### CODE REVIEW

File: lib/cloudkeeper/backend_connector.rb [hunk 1]
```diff
@@ -56,7 +57,7 @@ def convert_image(image)
       image_file = acceptable_image_file image
 
       Cloudkeeper::Grpc::Image.new mode: :LOCAL, location: image_file.file, format: image_file.format.upcase,
-                                   checksum: image_file.checksum, size: image.size, uri: image.uri
+                                   checksum: image_file.checksum, size: image.size.to_i, uri: image.uri
```
Comment (arax) [inline lib/cloudkeeper/backend_connector.rb]: Are you sure you want to implicitly convert `nil` values in `image.size` to `0`? This is not necessarily wrong ... you just need to be aware of that.

File: lib/cloudkeeper/backend_connector.rb [hunk 2]
```diff
@@ -67,12 +68,12 @@ def set_remote_data(image_proto, access_data)
     end
 
     def convert_appliance(appliance, image_proto)
-      Cloudkeeper::Grpc::Appliance.new identifier: appliance.identifier, description: appliance.description,
-                                       mpuri: appliance.mpuri, title: appliance.title, group: appliance.group,
-                                       ram: appliance.ram, core: appliance.core, version: appliance.version,
-                                       architecture: appliance.architecture, operating_system: appliance.operating_system,
-                                       vo: appliance.vo, image: image_proto, expiration_date: appliance.expiration_date.to_i,
-                                       image_list_identifier: appliance.image_list_identifier, attributes: appliance.attributes
+      Cloudkeeper::Grpc::Appliance.new identifier: appliance.identifier.to_s, description: appliance.description.to_s,
```
Comment (arax) [inline lib/cloudkeeper/backend_connector.rb]: The same applies here. Using `.to_s` and `.to_i` will convert `nil` values to empty strings and zeroes respectively. You will lose the difference between __not set__ and __set to emtpy/zero__.

File: lib/cloudkeeper/entities/appliance.rb [hunk 3]
```diff
@@ -73,7 +73,8 @@ def construct_os_name!(appliance, appliance_hash)
         end
 
         def populate_attributes!(appliance, appliance_hash)
-          appliance.attributes = appliance_hash.clone
+          appliance_hash.reject! { |k, _v| k == :vo || k == :expiration || k == :image_list_identifier }
```
Comment (arax) [inline lib/cloudkeeper/entities/appliance.rb]:
```ruby
REJECTED_ATTRIBUTES = [:vo, :expiration, :image_list_identifier].freeze
# ... HERE THERE BE DRAGONS ...
REJECTED_ATTRIBUTES.include? k
```

File: lib/cloudkeeper/entities/convertables/convertable.rb [hunk 4]
```diff
@@ -5,7 +5,12 @@ module Entities
     module Convertables
       module Convertable
         CONVERT_OUTPUT_FORMATS = [:raw, :qcow2, :vmdk].freeze
-        FORMAT_REGEX = /^to_(?<format>.*)$/
+
+        def self.convert_output_formats
+          CONVERT_OUTPUT_FORMATS
+        end
+
+        FORMAT_REGEX = Regexp.new "^to_(?<format>#{convert_output_formats.join('|')})$"
```
Comment (arax) [inline lib/cloudkeeper/entities/convertables/convertable.rb]: `Regexp` should be able to interpolate strings even when slashes are used.
```ruby
FORMAT_REGEX = /^to_(?<format>#{convert_output_formats.join('|')})$/
```

File: lib/cloudkeeper/backend_connector.rb (hunk 2 above)
Comment (Misenko) [inline lib/cloudkeeper/backend_connector.rb]: I am aware of that and it's ok.

**Pull request merged**